1. soru

In [1]:
import numpy as np
import pandas as pd
from sklearn import linear_model
import matplotlib.pyplot as plt
import plotly.graph_objs as go
import plotly.plotly as py
import scipy.stats as stats
import statsmodels.api as sm

import warnings
warnings.filterwarnings('ignore')

In [2]:
weather = pd.read_csv('weatherHistory.csv')

In [3]:
weather['temp_diff']=weather['Temperature (C)']- weather['Apparent Temperature (C)']

In [4]:
weather.head()

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   
3                  5.944444      0.83            14.1036   
4                  6.977778      0.83            11.0446   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   
3                   269.0          15.8263         0.0               1016.41   
4                   259.0          15.8263         0.0               1016.51   

                       Daily Summary  temp_diff  
0  Partly cloudy throughout the day.   2.083333  
1  Partly cloudy throughout the day.   2.127778  
2  Partly cloudy throughout the day.   0.000000  
3  Partly cloudy throughout the day.   2.344444  
4  Partly cloudy throughout the day.   1.777778

In [5]:
X=weather[['Humidity','Wind Speed (km/h)']]

In [6]:
Y=weather['temp_diff']

In [7]:
X = sm.add_constant(X)

In [8]:
sonuclar = sm.OLS(Y, X).fit()
sonuclar.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              temp_diff   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Sun, 26 May 2019   Prob (F-statistic):               0.00
Time:                        23:53:51   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -2.4381      0.021   -115.948      0.000      -2.479      -2.397
Humidity              3.0292      0.024    126.479      0.000       2.982       3.076
Wind Speed (km/h)     0.1193      0.001    176.164      0.000       0.118       0.121
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.264
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                           0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

b şıkkı

In [9]:
weather['Humidity_WindS']=weather['Humidity']*weather['Wind Speed (km/h)']

In [10]:
X=weather[['Humidity','Wind Speed (km/h)','Humidity_WindS']]

In [11]:
Y=weather['temp_diff']

In [12]:

X = sm.add_constant(X)

In [13]:
sonuclar = sm.OLS(Y, X).fit()
sonuclar.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              temp_diff   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Sun, 26 May 2019   Prob (F-statistic):               0.00
Time:                        23:56:25   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -0.0839      0.033     -2.511      0.012      -0.149      -0.018
Humidity             -0.1775      0.043     -4.133      0.000      -0.262      -0.093
Wind Speed (km/h)    -0.0905      0.002    -36.797      0.000      -0.095      -0.086
Humidity_WindS        0.2971      0.003     88.470      0.000       0.291       0.304
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.262
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                           0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

c şıkkı

In [15]:
X=weather[['Humidity','Wind Speed (km/h)','Visibility (km)']]

In [16]:
Y=weather['temp_diff']

In [17]:
X = sm.add_constant(X)

In [18]:
sonuclar = sm.OLS(Y, X).fit()
sonuclar.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              temp_diff   R-squared:                       0.304
Model:                            OLS   Adj. R-squared:                  0.303
Method:                 Least Squares   F-statistic:                 1.401e+04
Date:                Mon, 27 May 2019   Prob (F-statistic):               0.00
Time:                        00:00:30   Log-Likelihood:            -1.6938e+05
No. Observations:               96453   AIC:                         3.388e+05
Df Residuals:                   96449   BIC:                         3.388e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -1.5756      0.028    -56.605      0.000      -1.630      -1.521
Humidity              2.6066      0.025    102.784      0.000       2.557       2.656
Wind Speed (km/h)     0.1199      0.001    179.014      0.000       0.119       0.121
Visibility (km)      -0.0540      0.001    -46.614      0.000      -0.056      -0.052
==============================================================================
Omnibus:                     3833.895   Durbin-Watson:                   0.279
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4584.022
Skew:                           0.459   Prob(JB):                         0.00
Kurtosis:                       3.545   Cond. No.                         131.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

					
	F TEST	R2 TEST	ADJ R2 TEST	AIC	BIC
1. MODEL	1,95E+07	0.288	0.288	3,41E+08	3,41E+08
2. MODEL	1,67E+07	0.341	0.341	3,33E+08	3,33E+08
3. MODEL	1,40E+07	0.304	0.303	3,39E+08	3,39E+08

* AIC veBIC puanlarına göre 2. model çok daha iyidir.
* 2. model birinciye göre iyileşti
* adj R2 açısından ikinci model en iyisi , f testi acısından 3. model en iyisi

2. soru

Dep. Variable:	SalePrice	
* R-squared:	0.864
* Adj. R-squared:	0.853
* F-statistic:	80.53
* Prob (F-statistic):	0.00
* AIC:	2.587e+04
* BIC:	2.627e+04
		


Değerler güzel gözükmektedir.

6.3 odevınde tanımlanan ikinci modele gore (ki 1. modelin alt kumesıdır) 
Dep. Variable:	SalePrice	
* R-squared:	0.848
* Adj. R-squared:	0.840
* F-statistic:	103.4
* Prob (F-statistic):	0.00
* AIC:	2.594e+04
* BIC:	2.623e+04

ilk model daha ıyıdır.